test/e2e/persistent_volumes.go

```diff
@@ -374,17 +400,22 @@ func testPodSuccessOrFail(c clientset.Interface, ns string, pod *api.Pod) {
 
 // Delete the passed in pod.
 func deletePod(f *framework.Framework, c clientset.Interface, ns string, pod *api.Pod) {
+	if c != nil {
+		if pod != nil && len(pod.Name) > 0 {
+			framework.Logf("Deleting pod %v", pod.Name)
+			_, err := c.Core().Pods(ns).Get(pod.Name)
+			if !apierrs.IsNotFound(err) {
+				Expect(err).NotTo(HaveOccurred())
+				err := c.Core().Pods(ns).Delete(pod.Name, nil)
+				Expect(err).NotTo(HaveOccurred())
 
-	framework.Logf("Deleting pod %v", pod.Name)
-	err := c.Core().Pods(ns).Delete(pod.Name, nil)
-	Expect(err).NotTo(HaveOccurred())
-
-	// Wait for pod to terminate.  Expect apierr NotFound
-	err = f.WaitForPodTerminated(pod.Name, "")
-	Expect(err).To(HaveOccurred())
-	Expect(apierrs.IsNotFound(err)).To(BeTrue())
-
-	framework.Logf("Ignore \"not found\" error above. Pod %v successfully deleted", pod.Name)
+				// Wait for pod to terminate.  Expect apierr NotFound
+				err = f.WaitForPodTerminated(pod.Name, "")
+				Expect(err).To(HaveOccurred())
+				framework.Logf("Ignore \"not found\" error above. Pod %v successfully deleted", pod.Name)
```

Comment:
Ok, so do we care in the case where "not found" is logged yet that is the expected result? IOW, we want the pod to be deleted and not found, but we don't want someone reading the logs to think there is an error.